WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -456,26 +457,81 @@ async Task<(AliceClient? AliceClient, PersonCircuit? PersonCircuit)> RegisterInp
 
 		roundState.LogDebug($"Inputs({smartCoins.Count()}) registration started - it will end in: {remainingTimeForRegistration:hh\\:mm\\:ss}.");
 
-		// Decrease the available time, so the clients hurry up.
+		// TODO: Having variations in these values between clients could harm privacy, should be round parameters
+		var hurryUpThreshold = 5;
+		var hurryUpWindow = TimeSpan.FromMinutes(1);
 		var safetyBuffer = TimeSpan.FromMinutes(1);
-		var scheduledDates = GetScheduledDates(smartCoins.Count(), roundState.InputRegistrationEnd - safetyBuffer);
-
-		// Creates scheduled tasks (tasks that wait until the specified date/time and then perform the real registration)
-		var aliceClients = smartCoins.Zip(
-			scheduledDates,
-			async (coin, date) =>
+		
+		var nbInputsToRegister = smartCoins.Count();
+		var scheduledDates = GetScheduledDates(nbInputsToRegister, roundState.InputRegistrationEnd - safetyBuffer);
+
+		var aliceClients = new List<Task<(AliceClient? AliceClient, PersonCircuit? PersonCircuit)>>();
+		var index = 0;
+		var hurryMode = false;
+		while(index < scheduledDates.Count && roundState.Phase is Phase.InputRegistration && !registrationsCts.IsCancellationRequested)
```

Comment:
I am sure this logic works but it seems a bit complex. How about this ->

- Use the same logic that we have before with the Zip - in normal mode the code will look like the same, except
- Add one cancellationTokenSource: hurryUpCts and combine it into Task.Delay cancellation
- Instead of await Task.WhenAll(aliceClients), run a while loop and check if the hurry-up condition met
- If the hurry-up condition is met then cancel hurryUpCts
- After that repeat the same zip but with a shorter timeframe and use await Task.WhenAll

There would be two separate and visible step, first the normal operation, second fallback to hurry up mode.